encoding/marshal.go

```diff
@@ -17,10 +17,9 @@ type SiaMarshaler interface {
 	MarshalSia() []byte
 }
 
-// A SiaUnmarshaler can decode itself from a byte slice. If a decoding error
-// occurs, UnmarshalSia should panic.
+// A SiaUnmarshaler can decode itself from a byte slice.
 type SiaUnmarshaler interface {
-	UnmarshalSia([]byte)
+	UnmarshalSia([]byte) error
```

Comment:
that would imply reading the entire byte slice into memory before passing it to `Unmarshal`, which means you can't do any sort of streaming. I would prefer to pass a reader. Then `Currency` unmarshalling, for example, would look like:
```go
// UnmarshalSia implements the encoding.SiaUnmarshaler interface.
func (c *Currency) UnmarshalSia(r io.Reader) error {
	b, err := encoding.ReadPrefix(r, 256) // max currency length
	if err != nil {
		return err
	}
	c.i.SetBytes(b)
	return nil
}
```